File: JavaScript/0001multiples_of_3_or_5.ts
/*
 + Node 23.10.0
 * 0001multiples_of_3_or_5.ts
 * 
 * Copyright 2025 @GitHub JoergEm
 * 
 * This program is WITHOUT ANY WARRANTY; without even the implied
 * warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
 * 
 * 
 */

const multiples_of_3_or_5 = (limit: number): number => {
    return Array.from({ length: limit }, (_, index) => index)
        .filter(num => num % 3 === 0 || num % 5 === 0)
        .reduce((acc, num) => acc + num, 0);
};

const result: number = multiples_of_3_or_5(1000);
console.log(result);

File: Haskell/0001multiples_of_3_or_5.hs
{- 
 -- The Glorious Glasgow Haskell Compilation System, version 9.4.8
 -- 0001multiples_of_3_or_5.hs
 -- 
 -- Copyright 2025 @GitHub JoergEm
 -- 
 -- This program is WITHOUT ANY WARRANTY; without even the implied
 -- warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
  -}

summe :: Integer -> Integer -> Integer
summe n limit =
  let k = limit `div` n
  in n * k * (k + 1) `div` 2

main :: IO ()
main = print $ summe 3 999 + summe 5 999 - summe 15 999